File: tests/python/cases/case_177_dead_branch_global_08.py
"""
Task case 177: dead_branch_global_08 false-positive/false-negative stress pattern.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_177_dead_branch_global_08(input_data):
    task_name = 'dead_branch_global_08'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'stress',
        'stage': 'evaluation',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    if False:
        RETAINED_CACHE['case_177'] = payload
    # SAFE: sink exists only in dead branch and never executes.
    return payload['checksum']

File: tests/nodejs/cases/case_177_dead_branch_global_08.js
/**
 * Task case 177: dead_branch_global_08 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case177DeadBranchGlobal08(input) {
  const taskName = 'dead_branch_global_08';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  if (false) {
    retainedCache.set('case_177', payload);
  }
  // SAFE: sink is dead branch only.
  return payload.checksum;
}

module.exports = {
  case177DeadBranchGlobal08,
};
